amp/TEMPLATE/reampv2/src/modules/geocoder/components/layout/panel/GeocoderPanel.jsx

```diff
@@ -1,23 +1,101 @@
 import React, {Component} from 'react';
 import {TranslationContext} from '../../AppContext';
 import GeocoderHeader from "./GeocoderHeader";
-import TableBootstrap from "../table/ActivityTable";
+import ActivityTable from "../table/ActivityTable";
+import {bindActionCreators} from "redux";
+import {connect} from "react-redux";
+import GeocodingTable from "../table/GeocodingTable";
+import {SECTORS_CHART} from "../../../../sscdashboard/utils/constants";
 
-export default class GeocoderPanel extends Component {
+function GeocodingNotAvailable(props) {
+    return <h4>Geocoding process not available. User {props.user} is owner of the process in '{props.workspace}' workspace</h4>;
+}
+
+function GeocodingRunning() {
+    return (
+        <div>
+            <h4>Geocoding is still running</h4>
+        </div>
+    );
+}
+
+function ProjectList(props) {
+    return <h3>{props.title}</h3>;
+}
+
+class GeocoderPanel extends Component {
+
+    state = {
+        selectedActivities: []
+    };
+
+    onSelectActivity = (isSelected, activityId) => {
+        let selectedActivities = Array.from(this.state.selectedActivities);
+        if (isSelected) {
+            selectedActivities.push(activityId);
+        } else {
+            selectedActivities = selectedActivities.filter(id => id !== activityId);
+        }
+
+        this.setState({selectedActivities});
+    }
+
+    onSelectAllActivities = (isSelected, rows) => {
+        let selectedActivities = Array.from(this.state.selectedActivities);
+        let activityIds = rows.map(r => r.id);
+
+        if (isSelected) {
+            activityIds.forEach(activityId => selectedActivities.push(activityId));
+        } else {
+            selectedActivities = selectedActivities.filter(id => !activityIds.includes(id));
+        }
+
+        this.setState({selectedActivities});
```

Comment:
if you need to set state based on previous state you need to run all the code inside the set state method to ensure you get the last state. This way react guarantees that you are having the latest sate
`    onSelectActivity = (isSelected, activityId) => {
        this.setState(previousState => {
            let selectedActivities = Array.from(previousState.selectedActivities);
            if (isSelected) {
                selectedActivities.push(activityId);
            } else {
                selectedActivities = selectedActivities.filter(id => id !== activityId);
            }      
            return {selectedActivities};
        });
    }`

Reply:
intersting